src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const listOfStates = [];
+  const newState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(newState, action.extraData);
+        listOfStates.push({ ...newState });
+        break;
+      case 'removeProperties':
+        removeProperties(newState, action.keysToRemove);
+        listOfStates.push({ ...newState });
+        break;
+      case 'clear':
+        clear(newState);
+        listOfStates.push({});
```

Comment:
The 'clear' case should not push an empty object directly. Instead, it should modify the 'newState' to be an empty object and then push it after the switch statement to maintain consistency and avoid direct mutation.